python/cugraph/centrality/katz_centrality.py

```diff
@@ -0,0 +1,74 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.centrality import katz_centrality_wrapper
+
+
+def katz_centrality(G,
+                    alpha=0.1,
+                    max_iter=100,
+                    tol=1.0e-5,
+                    nstart=None,
+                    normalized=True):
+    """
+    Compute the Katz centrality for the nodes of the graph G
+
+    Parameters
+    ----------
+    graph : cuGraph.Graph
+        cuGraph graph descriptor with connectivity information. Edge weights,
+        if present, should be single or double precision floating point values.
```

Comment:
I suspect we temporarily set edge weights to something if they are not present? If so we may want to mention it here.

Reply:
I don't think that we need edges weights at all. Katz centrality doesn't use weights by default. So edge weights will be ignored.

Reply:
Sounds good, let's update the doc to reflect that.

Reply:
and if we're going to document hornet (gunrock, graphBLAS) algorithms as other cuGraph algorithms, we need to add

```
Centrality
=======

Katz Centrality
---------------

.. automodule:: cugraph.centrality.katz_centrality (or cugraph.cuhornet.centrality.katz_centrality based on where we place this in the python directory).
    :members:
    :undoc-members:
```

to

https://github.com/rapidsai/cugraph/blob/branch-0.10/docs/source/api.rst